### ISSUE TITLE
function as default field value 

### ISSUE THREAD

**Issue by puyaars** [eveningkid/denodb#198]
something like mongoose schema
passing a function that will get called when we create a new model instance

`class MyModel extends Model{
   static defaults = {
         date : Date.now
   }
}
`

**Comment by stillalivx** [comment]
Amm... I think that it is possible. In this pull request was implement. https://github.com/eveningkid/denodb/pull/147
I didn't work with that, but I think is something like the way you wrote it. No, @sntg-p?

**Comment by sntg-p** [comment]
Indeed, I made that pull request. That behavior you are looking for, @puyaars, is already implemented. I didn't do too much testing though as I stopped using this library, but feel free to add some context if the feature isn't working as you expect.

**Comment by eveningkid** [comment]
Hey,

I know it's been a long time already but I was really wondering why this wasn't working.

The issue is the following:

```js
// Model is User with:
// - name:string
// - age: int, default: Date.now

const user = new User();
user.name = "Test";
await user.save();
// user.age should exist

const nextUser = await User.create({ name: "Test" });
// nextUser.age won't exist
```

The reason is that the defaults [are only added when we call `.save`](https://github.com/sntg-p/denodb/blob/f93bfb6c68d2e7655e49d31f4c9c47665e0714ad/lib/model.ts#L761-L786), not with `.create`.

It would be great to take the logic from `.save` and create a method that takes values and turn them into values with defaults:

```js
const values = { name: "Test" };
const valuesWithDefaults = this._wrapValuesWithDefaults(values);
```

**Comment by eveningkid** [comment]
Alright, quick update on this: @stillalivx implemented the fix in #248 :)

All you need is upgrade to `1.0.37` now.

Thanks again for reporting this issue and have a great day!
